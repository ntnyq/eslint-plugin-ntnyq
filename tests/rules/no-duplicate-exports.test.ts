import { expect } from 'vitest'
import rule, { RULE_NAME } from '../../src/rules/no-duplicate-exports'
import { $, run } from '../internal'

run({
  name: RULE_NAME,
  rule,
  valid: [],
  invalid: [
    {
      description: 'export-all',
      filename: 'export-all.ts',
      code: $`
        export * from './foobar'
        export * from './foobar'
        
        export * as foobar from './foobar'
        export * as foobar from './foobar'
        
        export type * from './foobar'
        export type * from './foobar'
        
        export type * as Foobar from './foobar'
        export type * as Foobar from './foobar'
      `,
      errors(errors) {
        expect(errors).toMatchSnapshot('errors')
      },
      output(output) {
        expect(output).toMatchSnapshot('output')
      },
    },
    {
      description: 'named-export',
      filename: 'named-export.ts',
      code: $`
        export { foo, bar as baz } from './foobar'
        export { bar } from './foobar'
        export { type Bar } from './foobar'
        export type { Foo } from './foobar'
        export { default as 'module.exports' } from './foobar'
      `,
      errors(errors) {
        expect(errors).toMatchSnapshot('errors')
      },
      output(output) {
        expect(output).toMatchSnapshot('output')
      },
    },
  ],
})
