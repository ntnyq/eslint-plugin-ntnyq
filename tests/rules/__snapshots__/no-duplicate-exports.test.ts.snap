// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-duplicate-exports > invalid > Invalid #0: export-all > errors 1`] = `
[
  {
    "column": 1,
    "endColumn": 25,
    "endLine": 1,
    "line": 1,
    "message": "Multiple same export all statement 'export * from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 25,
    "endLine": 2,
    "fix": {
      "range": [
        25,
        49,
      ],
      "text": "",
    },
    "line": 2,
    "message": "Multiple same export all statement 'export * from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 35,
    "endLine": 3,
    "line": 3,
    "message": "Multiple same export all statement 'export * as foobar from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 35,
    "endLine": 4,
    "fix": {
      "range": [
        85,
        119,
      ],
      "text": "",
    },
    "line": 4,
    "message": "Multiple same export all statement 'export * as foobar from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 30,
    "endLine": 5,
    "line": 5,
    "message": "Multiple same export all statement 'export type * from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 30,
    "endLine": 6,
    "fix": {
      "range": [
        150,
        179,
      ],
      "text": "",
    },
    "line": 6,
    "message": "Multiple same export all statement 'export type * from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 40,
    "endLine": 7,
    "line": 7,
    "message": "Multiple same export all statement 'export type * as Foobar from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 40,
    "endLine": 8,
    "fix": {
      "range": [
        220,
        259,
      ],
      "text": "",
    },
    "line": 8,
    "message": "Multiple same export all statement 'export type * as Foobar from "./foobar"'",
    "messageId": "multiSameExportAll",
    "nodeType": "ExportAllDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
]
`;

exports[`no-duplicate-exports > invalid > Invalid #0: export-all > output 1`] = `
"export * from './foobar'

export * as foobar from './foobar'

export type * from './foobar'

export type * as Foobar from './foobar'
"
`;

exports[`no-duplicate-exports > invalid > Invalid #1: named-export > errors 1`] = `
[
  {
    "column": 1,
    "endColumn": 43,
    "endLine": 1,
    "fix": {
      "range": [
        0,
        42,
      ],
      "text": "export { foo, bar as baz, bar, type Bar, default as 'module.exports' } from './foobar'",
    },
    "line": 1,
    "message": "Multiple named export from same source './foobar'",
    "messageId": "multiSameSourceNamed",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 31,
    "endLine": 2,
    "fix": {
      "range": [
        43,
        73,
      ],
      "text": "",
    },
    "line": 2,
    "message": "Multiple named export from same source './foobar'",
    "messageId": "multiSameSourceNamed",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 36,
    "endLine": 3,
    "fix": {
      "range": [
        74,
        109,
      ],
      "text": "",
    },
    "line": 3,
    "message": "Multiple named export from same source './foobar'",
    "messageId": "multiSameSourceNamed",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
  {
    "column": 1,
    "endColumn": 55,
    "endLine": 5,
    "fix": {
      "range": [
        146,
        200,
      ],
      "text": "",
    },
    "line": 5,
    "message": "Multiple named export from same source './foobar'",
    "messageId": "multiSameSourceNamed",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "no-duplicate-exports",
    "severity": 2,
  },
]
`;

exports[`no-duplicate-exports > invalid > Invalid #1: named-export > output 1`] = `
"export { foo, bar as baz, bar, type Bar, default as 'module.exports' } from './foobar'


export type { Foo } from './foobar'
"
`;
